### PR TITLE
CDAP-16394 fix fll direction response

### PR DESCRIPTION
**Pull request by yaojiefeng**
https://issues.cask.co/browse/CDAP-16394

https://builds.cask.co/browse/CDAP-DUT7134

### CODE REVIEW

**Review by sreevatsanraman — CHANGES_REQUESTED** [review]
can we add a test?

**Review by nitinmotgi — COMMENTED** [review]
Shouldn't we find why we are seeing the null pointer.

**Comment by yaojiefeng** [comment]
This happens in the validation stage, essentially this direction query param must be provided by user, but previously we didn't check for that. We did that for the start and end time, we should do the same thing here.

**Review by sreevatsanraman — APPROVED** [review]

File: cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/LineageHTTPHandler.java [hunk 1]
```diff
@@ -301,7 +301,7 @@ public long getEnd() {
   private Constants.FieldLineage.Direction parseDirection(String directionStr) throws BadRequestException {
     try {
       return Constants.FieldLineage.Direction.valueOf(directionStr.toUpperCase());
-    } catch (IllegalArgumentException e) {
+    } catch (NullPointerException | IllegalArgumentException e) {
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/LineageHTTPHandler.java:304]: Is that mean `directionStr` can be null? Please annotate with `@Nullable` if that's the case.

Comment (yaojiefeng) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/LineageHTTPHandler.java:304]: Fixed

File: cdap-client/src/main/java/io/cdap/cdap/client/LineageClient.java [hunk 2]
```diff
@@ -131,10 +132,31 @@ public LineageRecord getLineage(DatasetId datasetInstance, String startTime, Str
...
+   * @return {@link FieldLineageSummary} for the specified dataset.
+   */
+  public FieldLineageSummary getFieldLineage(DatasetId datasetId, long startTime, long endTime,
+                                             String direction)
+    throws IOException, BadRequestException, NotFoundException, UnauthenticatedException {
+    String path = String.format("datasets/%s/lineage/allfieldlineage?start=%s&end=%s", datasetId.getDataset(),
+                                URLEncoder.encode(Long.toString(startTime), "UTF-8"),
+                                URLEncoder.encode(Long.toString(endTime), "UTF-8"));
+    if (direction != null) {
+      path = String.format("%s&direction=%s", path, direction);
```
Comment (chtyim) [inline cdap-client/src/main/java/io/cdap/cdap/client/LineageClient.java]: Strictly speaking, should have `URLEncoder.encode(direction)`.

**Pull request merged**